python/llm/README.md

```diff
@@ -1,140 +1,137 @@
 
-Here is an example to use `llm-convert` command line tool.
-```bash
-# pth model
-llm-convert "/path/to/llama-7b-hf/" --model-format pth --outfile "/path/to/llama-7b-int4/" --model-family "llama"
-# gptq model
-llm-convert "/path/to/vicuna-13B-1.1-GPTQ-4bit-128g/" --model-format gptq --outfile "/path/to/vicuna-13B-int4/" --model-family "llama"
-```
-> An example GPTQ model can be found [here](https://huggingface.co/TheBloke/vicuna-13B-1.1-GPTQ-4bit-128g/tree/main)
+### Demos
+See the ***optimized performance*** of `phoenix-inst-chat-7b`, `vicuna-13b-v1.3`, and `starcoder-15b` models on a 12th Gen Intel Core CPU below.
 
-Here is an example to use `llm_convert` python API.
-```bash
-from bigdl.llm import llm_convert
-# pth model
-llm_convert(model="/path/to/llama-7b-hf/",
-            outfile="/path/to/llama-7b-int4/",
-            model_format="pth",
-            model_family="llama")
-# gptq model
-llm_convert(model="/path/to/vicuna-13B-1.1-GPTQ-4bit-128g/",
-            outfile="/path/to/vicuna-13B-int4/",
-            model_format="gptq",
-            model_family="llama")
-```
+[Placeholder for screen video capture]
 
-### Inferencing
-
-#### llm-cli command line
-llm-cli is a command-line interface tool that follows the interface as the main program in `llama.cpp`.
+### Working with `bigdl-llm`
 
+#### Install
+You may install **`bigdl-llm`** as follows:
 ```bash
-# text completion
-llm-cli -t 16 -x llama -m "/path/to/llama-7b-int4/bigdl-llm-xxx.bin" -p 'Once upon a time,'
-
-# chatting
-llm-cli -t 16 -x llama -m "/path/to/llama-7b-int4/bigdl-llm-xxx.bin" -i --color
-
-# help information
-llm-cli -x llama -h
-```
-
-#### Transformers like API
-You can also load the converted model using `BigdlForCausalLM` with a transformer like API, 
-```python
-from bigdl.llm.transformers import BigdlForCausalLM
-llm = BigdlForCausalLM.from_pretrained("/path/to/llama-7b-int4/bigdl-llm-xxx.bin",
-                                           model_family="llama")
-prompt="What is AI?"
-```
-and simply do inference end-to-end like
-```python
-output = llm(prompt, max_tokens=32)
-```
-If you need to seperate the tokenization and generation, you can also do inference like
-```python
-tokens_id = llm.tokenize(prompt)
-output_tokens_id = llm.generate(tokens_id, max_new_tokens=32)
-output = llm.batch_decode(output_tokens_id)
+pip install --pre --upgrade bigdl-llm[all]
 ```
-
-
-Alternatively, you can load huggingface model directly using `AutoModelForCausalLM.from_pretrained`. 
+#### Download Model
+
+You may download any PyTorch model in Hugging Face *Transformers* format (including *FP16* or *FP32* and *GPTQ-4bit*).
+
+#### Run Model
+ 
+You may run the models using **`bigdl-llm`** through one of the following APIs:
+1. [CLI (command line interface) Tool](#cli-tools)
+2. [Hugging Face `transformer`-style API](#hugging-face-transformers-style-api)
+3. LangChain API
+4. `llama-cpp-python`-style API
+
+#### CLI Tools
```

Comment:
please verify the convert and cli command @hoshibara @leonardozcm

Reply:
> please verify the convert and cli command @hoshibara @leonardozcm

llm-convert works fine after https://github.com/intel-analytics/BigDL/pull/8395